servers/rendering/rendering_server_default.h

```diff
@@ -313,7 +313,16 @@ class RenderingServerDefault : public RenderingServer {
 	FUNC2(mesh_set_path, RID, const String &)
 	FUNC1RC(String, mesh_get_path, RID)
 
-	FUNC2(mesh_set_shadow_mesh, RID, RID)
+	virtual void mesh_set_shadow_mesh(RID p1, RID p2) override {
+		ERR_FAIL_COND_MSG(p1 == p2, "Cannot put the mesh itself as its own shadow mesh.");
```

Comment:
This should be added to the implemented of `mesh_set_shadow_mesh` instead of adding a shim here.